server.js

```diff
@@ -0,0 +1,246 @@
+        return res.end();
+      };
+      if (req.method === 'GET') {
+        //on GET reads file and sends content back as body, 404 return if it does not exist
+        fs.readFile('./public' + path, 'utf8', function (err, data) {
+          if (err && err.code === 'ENOENT') {
+            res.statusCode = 404;
+            fs.readFile('./public/404.html', 'utf8', function (err, data) {
+              if (err) { throw err; };
+              return res.end(data);
+            });
+          } else if (err) {
+            console.log(err);
+            throw err;
+          } else {
+            res.statusCode = 200;
+            return res.end(data);
+          };
+        });
+      } else if (req.method === 'POST') {
+        //on POST checks authorization and correct path for creation, then writes new page
+        if (!req.headers.authorization) {
+          res.writeHead(401, {
+            'WWW-Authenticate': 'Basic realm="Secure Area"'
+          });
+          return res.end('<html><body>Not Authorized</body></html>');
+        } else {
+          let encodedString = req.headers.authorization.split(' ')[1];
+          //check if authorization value was created correctly by client
+          if (typeof encodedString !== 'string') {
+            res.writeHead(401, {
+              'WWW-Authenticate': 'Basic realm="Secure Area"'
+            });
+            return res.end('<html><body>Not Authorized</body></html>');
+          };
+          let base64Buffer = Buffer.from(encodedString, 'base64');
+          let decodedString = base64Buffer.toString();
+
+          if (!authorize(decodedString)) {
+            res.writeHead(401, {
+              'WWW-Authenticate': 'Basic realm="Secure Area"'
+            });
+            return res.end('<html><body>Not Authorized</body></html>');
+          };
+        };
+        //check path
+        if (!(req.url === '/elements')) {
+          res.writeHead(400, {
+            'Content-Type': 'application/json'
+          });
+          return res.end('You cannot send a post to ' + req.url);
+        };
+        // check if file already exists, else write file
+        fs.stat('./public/' + elementData.elementName + '.html', function (err) {
+          if (err && err.code !== 'ENOENT') {
+            console.log(err);
+            throw err;
+          } else if (!err) {
+            res.writeHead(500, {
+              'Content-Type': 'application/json'
+            });
+            return res.end('{ "error" : "resource ' + elementData.elementName + '.html' + ' already exists" }');
+          } else {
+
+            fs.writeFile('./public/' + elementData.elementName + '.html', element(elementData), function (err) {
+              if (err) {
+                res.writeHead(500, {
+                  'Content-Type': 'application/json'
+                });
+                return res.end('{ "error" : "could not write file" }');
+              };
+              index();
```

Comment:
You should make your index take in the res.writeHead below and the return res.end() so that the redirect happens after you're finished re-writing your index.